MetaMask/metamask-extension
# fix: replace last usages of legacy gas fee polling with polling by networkClientId

## **Description**

https://github.com/MetaMask/metamask-extension/pull/23010 made initial changes to replace the GasFeeController's legacy gas fee polling in favor of polling via networkClientId. It was discovered that there are still lingering instances of the legacy gas fee polling which were causing double polling to occur since the legacy and networkClientId based polling run on separate loops. This PR fixes this by replacing those remaining legacy gas fee polling usages.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/24065?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/pull/23010

## **Manual testing steps**

1. Open background console -> Network tab
2. Open Extension Popup -> send -> enter address and amount -> next
3. Background console network tab should now start showing one single network request every 10s to the gas api endpoint


## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
